enterprise_access/apps/bffs/checkout/handlers.py

```diff
@@ -255,3 +275,168 @@ def _check_user_existence(self, email):
         except Exception:  # pylint: disable=broad-except
             # In case of error, we don't know if the user exists
             return None
+
+
+class CheckoutSuccessHandler(CheckoutContextHandler):
+    """
+    Handler for checkout success operations. Builds on the ``CheckoutContextHandler``
+    to enhance the checkout intent record with addtional data from the stripe API.
+    """
+    context: CheckoutSuccessContext
+
+    def load_and_process(self):
+        """
+        Loads base checkout context data, then enhances
+        the checkout intent record with more data from the Stripe API.
+        """
+        super().load_and_process()
+        if self.context.checkout_intent is None:
+            return
+
+        self.context.checkout_intent['first_billable_invoice'] = None
+
+        try:
+            self.enhance_with_stripe_data(self.context.checkout_intent)
+        except Exception as exc:  # pylint: disable=broad-exception-caught
+            logger.exception(
+                "Error loading checkout success handler Stripe data for request user %s",
+                self.context.user,
+            )
+            self.add_error(
+                user_message="Could not load and/or process checkout success data",
+                developer_message=f"Unable to load and/or process checkout success data: {exc}",
+            )
+
+    # pylint: disable=too-many-statements
+    def enhance_with_stripe_data(self, checkout_intent_data):
```

Comment:
ah, I actually don't need a parameter here, I can access `self.context.checkout_intent` directly within the body. will clean that up.